### PR TITLE
[Console] fix table setHeaderTitle without headers

### LINKED ISSUE #42196: [Console] Cannot set header title without headers row
**Symfony version(s) affected**: 5.3.3

**Description**
If headers row doesn't present in a Table, the `setHeaderTitle` call doesn't work.

**How to reproduce**
```php
(new Table($output))
    ->setHeaderTitle('Reproducer')
    ->setRows([
        ['Value', '123-456'],
        ['Some other value', '789-0'],
    ])
    ->render()
;
```
renders
```
+------------------+---------+
| Value            | 123-456 |
| Some other value | 789-0   |
+------------------+---------+
```

But
```php
(new Table($output))
    ->setHeaderTitle('Reproducer')
    ->setHeaders(['Key', 'Value'])
    ->setRows([
        ['Value', '123-456'],
        ['Some other value', '789-0'],
    ])
    ->render()
;
```
renders
```
+-------- Reproducer --------+
| Key              | Value   |
+------------------+---------+
| Value            | 123-456 |
| Some other value | 789-0   |
+------------------+---------+
```

### PR DESCRIPTION
**Pull request by a1812**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #42196 
| License       | MIT
| Doc PR        | 

fix bug

### CODE REVIEW

File: src/Symfony/Component/Console/Helper/Table.php [hunk 1]
```diff
@@ -374,6 +374,7 @@ public function render()
 
         $isHeader = !$this->horizontal;
         $isFirstRow = $this->horizontal;
+        $isTitleNeed = (bool) $this->headerTitle;
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Helper/Table.php]: $needsTitle ?

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 2]
```diff
@@ -1115,6 +1115,53 @@ public function renderSetTitle()
         ];
     }
 
+    /**
+     * @dataProvider renderSetTitle2
+     */
+    public function testSetTitle2($headerTitle, $headers, $expected)
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]:
```suggestion
    public function testSetTitle2(string $headerTitle, array $headers, string $expected)
```

File: src/Symfony/Component/Console/Helper/Table.php (hunk 1 above)
Comment (a1812) [reply src/Symfony/Component/Console/Helper/Table.php]: ok

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php (hunk 2 above)
Comment (a1812) [reply src/Symfony/Component/Console/Tests/Helper/TableTest.php]: ok

**Comment by a1812** [comment]
Hi @OskarStark, what do you think about reworking the test "testSetTitle" and include testSetTitle2 in it ?

**Comment by OskarStark** [comment]
IMHO the second test case in your new data provider is already included in the `testSetTitle` data provider.

So a test case `testSetTitleWithoutHeaders` without a data provider would be my favorite

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 3]
```diff
@@ -1115,6 +1115,53 @@ public function renderSetTitle()
...
+        (new Table($output = $this->getOutputStream()))
+            ->setHeaderTitle($headerTitle)
+            ->setHeaders($headers)
+            ->setRows([
+                ['Value', '123-456'],
+                ['Some other value', '789-0'],
+            ])
+            ->render();
+
+        $this->assertEquals($expected, $this->getOutputContent($output));
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]: `assertSame` whenever possible

File: src/Symfony/Component/Console/Helper/Table.php [hunk 4]
```diff
@@ -374,6 +374,7 @@ public function render()
 
         $isHeader = !$this->horizontal;
         $isFirstRow = $this->horizontal;
+        $needsTitle = (bool) $this->headerTitle;
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Helper/Table.php]: `hasTitle` sounds better, sorry for changing my mind

**Review by OskarStark — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @a1812.

**Pull request merged**